File: 20141123_CLASS_of_addressBook_by_Cplus/20141123_CLASS_of_addressBook_by_Cplus/personinfo.cpp
/*********************
* Name : personinfo.cpp
* Date : 2014.11.23
* Writer : Sejoong Kim
**********************/

#include <iostream>
#include <iomanip>
#include "personinfo.h"

//default constructor
PersonalInfo::PersonalInfo() {
	this->name = "";
	this->phone = "";
	this->address = "";
}

//copy constructor
PersonalInfo::PersonalInfo(PersonalInfo &info) {
	this->name = info.name;
	this->phone = info.phone;
	this->address = info.address;
}

// destructor
PersonalInfo::~PersonalInfo() {
	this->_name = "";
	this->_phone = "";
	this->_address = "";
}

//overloading operator = 
PersonalInfo& PersonalInfo::operator= (PersonalInfo& info) {
	this->name = info.name;
	this->phone = info.phone;
	this->address = info.address;

	return *this;
}

//overloading operator ==
bool PersonalInfo::operator== (PersonalInfo& info) {
	if (this->name == info.name) {
		return true;
	}
	else {
		return false;
	}	
}

//overloading operator ==
bool PersonalInfo::operator== (string& info) {
	if (this->name == info) {
		return true;
	}
	else {
		return false;
	}
}

//overloading operator <<
/*
ostream& PersonalInfo::operator<< (ostream& os) {
	os << os.width(8) << left << name << "  ";
	os << os.width(12) << left << phone << "  ";
	os << os.width(50) << left << address;
	return os;
}
*/

//overloading operator <<
ostream& operator<< (ostream& os, PersonalInfo& info) {
	/*
	os.width(8);
	os << left << info.name << "  ";
	os.width(12);
	os << left << info.phone << "  ";
	os.width(50);
	os << left << info.address;
	*/
	os << setw(8) << left << info.name << "  ";
	os << setw(12) << left << info.phone << "  ";
	os << setw(50) << left << info.address;

	return os;
}

// need to definition for accessing private variables
void PersonalInfo::setName(string name) {
	this->name = name;
}

string PersonalInfo::getName() {
	return this->name;
}

void PersonalInfo::setPhone(string phone) {
	this->phone = phone;
}

string PersonalInfo::getPhone() {
	return this->phone;
}

void PersonalInfo::setAddress(string address) {
	this->address = address;
}

string PersonalInfo::getAddress() {
	return this->address;
}

File: 20150228_addressBook_unittest_python/personInfo.py
# -*- coding: utf-8 -*-

"""
Name : main.cpp
Date : 2015.01.23
Writer : Sejoong Kim
"""

class personalInfo:
    #constructor
    def __init__(self, info = None):
        self._name = info._name if info else ""
        self._phone = info._phone if info else ""
        self._address = info._address if info else ""

    #Destructor
    def __del__(self):
        self._name = ""
        self._phone = ""
        self._address = ""

    #Built-in Functions
    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, value):
        self._name = value

    @property
    def phone(self):
        return self._phone

    @phone.setter
    def phone(self, value):
        self._phone = value

    @property
    def address(self):
        return self._address

    @address.setter
    def address(self, value):
        self._address = value

    #comparison
    def __eq__(self, other):
        isEqual = False

        if type(other) is personalInfo:
            if (self.name == other.name):
                isEqual = True
        elif type(other) is str:
            if (self.name == other):
                isEqual = True
        else:
            isEqual = False
        return isEqual